sos/report/plugins/opencontrail.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2021 Mirntis, Inc., Oleksii Molchanov <omolchanov@mirantis.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+
+
+class OpenContrail(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+    """OpenContrail SDN
+    """
+    plugin_name = 'opencontrail'
+    profiles = ('network')
+
+    def setup(self):
+        container_status = self.get_command_output(
+            "docker ps -a --format='{{ .Names }}'")
+
+        in_container = False
+        container_names = []
+        if container_status['status'] == 0:
+            for line in container_status['output'].splitlines():
+                if line.startswith("opencontrail"):
+                    in_container = True
+                    container_names.append(line)
+
+        if in_container:
+            for container in container_names:
+                self.add_cmd_output(
+                    'docker exec -t {0} contrail-status'
+                    .format(container)
+                )
```

Comment:
There is existing container plumbing within the Plugin API for this.

```python
in_container = self.container_exists('opencontrail.*')   # assuming the container names will start with "opencontrail"
if in_container:
    cnames = self.get_containers(get_all=True)
    cnames = [c[1] for c in cnames if 'opencontrail' in c[1]]
    for cntr in cnames:
        _cmd = self.fmt_container_cmd(cntr, 'contrail-status')
        self.add_cmd_output(_cmd)
```

This removes the need to manually construct and run commands for specific container runtimes - the loaded policy will determine which container runtime (docker, podman, etc...) to use.

Reply:
Fixed with provided example.